chia/full_node/full_node.py

```diff
@@ -145,7 +145,7 @@ class FullNode:
     #       config would end up on stdout if handled here.
     multiprocessing_context: Optional[BaseContext] = None
     _ui_tasks: Set[asyncio.Task[None]] = dataclasses.field(default_factory=set)
-    subscriptions: PeerSubscriptions = dataclasses.field(default_factory=PeerSubscriptions)
+    _subscriptions: Optional[PeerSubscriptions] = None
```

Comment:
why are you making this field optional?
it looks like it would add additional complexity. Especially the pre-condition to `subscriptions()` is now much more complex, since it asserts that this field is not `None`.

Reply:
I see, this is the convention now that we use `managed()`